contracts/strategy/BnbxYieldConverterStrategy.sol

```diff
@@ -0,0 +1,194 @@
+    }
+
+    /// @dev internal function to deposit the given amount of BNB into Stader stakeManager through bnbxRouter
+    /// @param amount amount of BNB
+    function _deposit(uint256 amount) internal returns (uint256 value) {
+        require(!depositPaused, "deposits are paused");
+        require(amount > 0, "invalid amount");
+        if (canDeposit(amount)) {
+            bnb_deposit_balance += amount;
+            bnbx_holding_balance += _stakeManager.convertBnbToBnbX(amount);
+            _stakeManager.deposit{value: amount}();
+        }
+    }
+
+    /// @dev withdraws the given amount of BNB from Stader stakeManager and transfers to masterVault
+    /// @param amount amount of BNB
+    function withdraw(address recipient, uint256 amount) onlyVault external returns(uint256 value) {
+        return _withdraw(recipient, amount);
+    }
+
+    /// @dev withdraws everything from Stader stakeManager and transfers to masterVault
+    function panic() external onlyStrategist returns (uint256 value) {
+        (,, uint256 debt) = vault.strategyParams(address(this));
+        return _withdraw(address(vault), debt);
+    }
+
+    /// @dev internal function to withdraw the given amount of BNB from Stader stakeManager
+    ///      and transfers to masterVault
+    /// @param amount amount of BNB
+    /// @return value - returns the amount of BNB withdrawn from stader
+    function _withdraw(address recipient, uint256 amount) internal returns (uint256 value) {
+        require(amount > 0, "invalid amount");
+
+        uint256 bnbxAmount = _stakeManager.convertBnbToBnbX(amount);
+        
+        bnb_deposit_balance -= amount;
+        bnbx_holding_balance -= bnbxAmount;
+        
+        userRequests[recipient].push( UserRequest({
+            batchId: nextWithdrawBatchId,
+            bnbAmt : amount,
+            bnbxAmount,
+            triggerTime,
+        }));
+
+        // or maybe queue userRequests;
+        // userRequests.push( UserRequest({
+        //     batchId: nextWithdrawBatchId,
+        //     bnbAmt : amount,
+        //     bnbxAmount,
+        //     triggerTime,
+        // }));
+
+        batchWithdrawTotal += bnbxAmount;
+
+        return 0;
+    }
+
+    function batchWithdraw(){
+        uint256 nextStakeManagerWithdrawId = _stakeManager.nextUndelegateUUID();
+        batchRequest[nextWithdrawBatchId] = BatchRequest({
+            nextStakeManagerWithdrawId,
+            batchWithdrawTotal,
+            triggerTime,
+            claimTime: 0
+        });
+        
+        _stakeManager.requestWithdraw(batchWithdrawTotal);
+
+        nextWithdrawBatchId++;
+        batchWithdrawTotal = 0;
+    }
+
+    function batchClaimWithdraw(){ // transfer funds(BNB) from stakeManager to strategy
+        batchWithdrawRequests = _stakeManager.getUserWithdrawalRequests();
+
+        for( each batchReqId : batchWithdrawRequests){ // would be max 7 as once per day
+            if batchReqId is claimable => claim it;
+        }
```

Comment:
does seem a proper way to mark batchRequests which are already claimed in this step